src/sentry/tasks/backfill_seer_grouping_records.py

```diff
@@ -155,45 +155,55 @@ def backfill_seer_grouping_records(
         )
 
     group_id_batch = list(group_id_message_batch_filtered.keys())
-    time_now = datetime.now()
     events_entity = Entity("events", alias="events")
     redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
 
-    query = Query(
-        match=events_entity,
-        select=[
-            Column("group_id"),
-            Function("max", [Column("event_id")], "event_id"),
-        ],
-        groupby=[Column("group_id")],
-        where=[
-            Condition(Column("project_id"), Op.EQ, project.id),
-            Condition(Column("group_id"), Op.IN, group_id_batch),
-            Condition(
-                Column("timestamp", entity=events_entity), Op.GTE, time_now - timedelta(days=90)
-            ),
-            Condition(Column("timestamp", entity=events_entity), Op.LT, time_now),
-        ],
-        orderby=[OrderBy(Column("group_id"), Direction.ASC)],
-    )
+    # TODO(jangjodi): Only query per group if it has over 1 million events, or batch queries with new where condition
+    snuba_requests = []
+    for group_id in group_id_batch:
```

Comment:
How big is a batch btw? We might want to be careful that we're not slamming snuba with too many queries at once, or we could hit a rate limit.

Reply:
the batch size is 20 and I've taken your suggestion to combine the call into a bulk query (with up to 20 requests)

Reply:
Right, that batch thing just makes 20 queries in a thread pool, so it's still a bunch of queries at once. The thread pool has 10 workers https://github.com/getsentry/sentry/blob/cebec75409806ecb190b1a6db68091b53d42d0c7/src/sentry/utils/snuba.py#L469

So as long as that doesn't put us over the limits for this referrer we should be ok

Reply:
I see, makes sense. Based on this [comment](https://github.com/getsentry/snuba/pull/5774#issuecomment-2057160235) we have 4 concurrent queries and two threads, but I'll ask about non-current but very close together queries

Reply:
The batch query running will run them concurrently using the thread pool. I'd probably recommend breaking into batches of 4 instead of 20

Reply:
I've decided to change the batch size to 4 for now, and then follow up to do the queries in batches of 4. Unfortunately I can't do the `(group.id = 1 and group.last_seen - 5 mins <= timestamp < group.last_seen + 5 mins) or ...` since the query always needs an an absolute time range provided

Reply:
That's a good point... Actually, I guess rather than the weird or statement, it'd be better to just do:

`group_id in (<group_ids>) and min(groups, key=lambda g: g.last_seen) - 5 mins <= timestamp and timestamp <= max(groups, key=lambda g: g.last_seen) + 5 mins`